src/platform/intel/cavs/include/cavs/lib/pm_memory.h

```diff
@@ -154,4 +154,27 @@ void cavs_pm_memory_lp_sram_power_gate(void *ptr, uint32_t size, bool enabled);
 
 #endif /* CONFIG_LP_SRAM */
 
+#if CONFIG_L1_DRAM
+
+static inline void cavs_pm_memory_l1_dram_banks_power_gate(
+	uint32_t start_bank_id, uint32_t ending_bank_id, bool enable)
+{
+	uint32_t mask = MASK(start_bank_id, ending_bank_id);
+	uint32_t pgctl = io_reg_read(L1_MEM_DRAM_PGCTL);
+	uint32_t pgists = io_reg_read(L1_MEM_DRAM_PGISTS);
+	uint32_t expected = enable ? pgists & mask : pgists | ~mask;
+
+	if (enable)
+		pgctl &= mask;
+	else
+		pgctl |= ~mask;
```

Comment:
apparently this works, bit it looks strange. We're sure it isn't inverted, i.e. it shouldn't be `x | mask` and `x & ~mask` in both cases above? So, say, if you want to enable bits 0 and 1. `mask = 3`. Let's say both `pgctl` and `pgists` read 1 (the first bank currently enabled?). Then you calculate `expected = 1; pgctl = 1;` i.e. the same as they were before - no change?

Reply:
Please keep in mind the logic is inverted here, a set bit means bank is power gated. I think the interface name is misleading here but I just stick to the standard of already defined..

Reply:
I have just pushed a change which shall make it more clear